src/hooks/useSignForWalletDerivation.tsx

```diff
@@ -28,5 +32,21 @@ export default function useSignForWalletDerivation() {
       }),
     [signTypedData, signTypedDataAsync, chainId]
   );
-  return callSignTypedData;
+
+  const { signMessage: phantomSignMessage } = usePhantomWallet();
+
+  const signSolanaMessage = useCallback(async (): Promise<string> => {
+    const signature = await phantomSignMessage(stableStringify(signTypedData));
+    // Left pad the signature with a 0 byte so that the signature is 65 bytes long, a solana signature is 64 bytes by default.
+    return Buffer.from([0, ...signature]).toString('hex');
+  }, [phantomSignMessage, signTypedData]);
+
+  const signMessage = useCallback(async (): Promise<string> => {
```

Comment:
[q] does this fn need to be async? since we're not awaiting anything inside

Reply:
the async keyword could be removed but it is still an "async" function since that function returns a promise.

Another way to write this function would be:
```
  const signMessage = useCallback(async (): Promise<string> => {
    if (walletType === WalletType.Phantom) {
      return await signSolanaMessage();
    }
    return await signEvmMessage();
  }, [signEvmMessage, signSolanaMessage, walletType]);
```
But await can be omitted in `return await` statements.

I think its still good to leave the async keyword there so that any uncaught exceptions would still result in a promise rejection.

Reply:
so is the benefit here that the app won't crash since it's wrapped in a promise? 

i may be misunderstanding here but that feels a little strange - if the function has an uncaught exception outside of asynchronous code then the expected behavior is that the app crashes, no? 

but if we expect that this function may throw an exception, then we can wrap the contents in a try/catch block to handle it specifically so we can get proper logging etc.

feels weird to 'hide' uncaught exceptions in a promise wrapper

Reply:
I guess all i am saying is `async () => {return f()}` and `() => {return f()}` is not equivalent and the former is what we want in this case.